### PR TITLE
[new release] bimage-unix, conf-openimageio, bimage, bimage-io, bimage-display and bimage-lwt (0.3.0)

### PR DESCRIPTION
**Pull request by zshipko**
Bimage_unix provides methods for encoding/decoding images in many formats using ImageMagick/stb_image

- Project page: <a href="https://github.com/zshipko/ocaml-bimage">https://github.com/zshipko/ocaml-bimage</a>
- Documentation: <a href="https://zshipko.github.io/ocaml-bimage/doc">https://zshipko.github.io/ocaml-bimage/doc</a>

##### CHANGES:

- Overhaul of `Color` and `Type`
- Added `FILTER`/`Filter`
- Added `Bimage_io` and `Bimage_lwt`

### CODE REVIEW

**Comment by zshipko** [comment]
I'm hoping to release this `conf-openimageio` package but I'm not sure if I set it up correctly - any suggestions to help fix that would be appreciated!

File: packages/conf-openimageio/conf-openimageio.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  ["openimageio"] {os = "macos" & os-distribution = "homebrew"}
+  ["openimageio"] {os = "macos" & os-distribution = "macports"}
+]
+synopsis: "Virtual package relying on OpenImageIO development package installation"
+description: """
+This package can only install if OpenImageIO is available on the system.
+"""
+flags: conf
+x-commit-hash: "683ba69b9a9c0406a85b4084da38c2dd604e390f"
+url {
```
Comment (mseri) [inline packages/conf-openimageio/conf-openimageio.0.3.0/opam:26]: You just need to remove this part. Conf packages carry no source. I can do it from here

Comment (zshipko) [reply packages/conf-openimageio/conf-openimageio.0.3.0/opam]: Thanks!

**Comment by zshipko** [comment]
Looks like there are some issues with the version of `OpenImageIO` provided and what I'm checking for (in addition to some other dependency weirdness) - I will have to keep looking into this a bit more!

File: packages/conf-openimageio/conf-openimageio.1/files/test.cpp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include <OpenImageIO/version.h>
+
+#if OIIO_VERSION_MAJOR >= 2
```
Comment (zshipko) [inline packages/conf-openimageio/conf-openimageio.1/files/test.cpp:3]: Should this be `OIIO_VERSION_MAJOR < 2`?

Comment (kit-ty-kate) [reply packages/conf-openimageio/conf-openimageio.1/files/test.cpp]: Indeed. My bad. Fixed.

**Comment by kit-ty-kate** [comment]
On Debian 11 (testing) and Sid (unstable), `bimage-io` fails with:
```
#=== ERROR while compiling bimage-io.0.3.0 ====================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/zshipko/ocaml-bimage/releases/download/v0.3.0/bimage-v0.3.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/bimage-io.0.3.0
# command              ~/.opam/4.11/bin/dune build -p bimage-io -j 47
# exit-code            1
# env-file             ~/.opam/log/bimage-io-15-803e1a.env
# output-file          ~/.opam/log/bimage-io-15-803e1a.out
### output ###
#          gcc io/input.o (exit 1)
# (cd _build/default/io && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -g -I /home/opam/.opam/4.11/lib/ocaml -I /home/opam/.opam/4.11/lib/bimage -I /home/opam/.opam/4.11/lib/ocaml/threads -o input.o -c input.cpp)
# In file included from /usr/include/OpenImageIO/fmath.h:40,
#                  from /usr/include/OpenImageIO/imagebuf.h:16,
#                  from io.h:5,
#                  from input.cpp:1:
# /usr/include/OpenImageIO/Imath.h:13:10: fatal error: OpenEXR/OpenEXRConfig.h: No such file or directory
#    13 | #include <OpenEXR/OpenEXRConfig.h>
#       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
#          gcc io/output.o (exit 1)
# (cd _build/default/io && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -g -I /home/opam/.opam/4.11/lib/ocaml -I /home/opam/.opam/4.11/lib/bimage -I /home/opam/.opam/4.11/lib/ocaml/threads -o output.o -c output.cpp)
# In file included from /usr/include/OpenImageIO/fmath.h:40,
#                  from /usr/include/OpenImageIO/imagebuf.h:16,
#                  from io.h:5,
#                  from output.cpp:1:
# /usr/include/OpenImageIO/Imath.h:13:10: fatal error: OpenEXR/OpenEXRConfig.h: No such file or directory
#    13 | #include <OpenEXR/OpenEXRConfig.h>
#       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
#          gcc io/spec.o (exit 1)
# (cd _build/default/io && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -g -I /home/opam/.opam/4.11/lib/ocaml -I /home/opam/.opam/4.11/lib/bimage -I /home/opam/.opam/4.11/lib/ocaml/threads -o spec.o -c spec.cpp)
# In file included from /usr/include/OpenImageIO/fmath.h:40,
#                  from /usr/include/OpenImageIO/imagebuf.h:16,
#                  from io.h:5,
#                  from spec.cpp:1:
# /usr/include/OpenImageIO/Imath.h:13:10: fatal error: OpenEXR/OpenEXRConfig.h: No such file or directory
#    13 | #include <OpenEXR/OpenEXRConfig.h>
#       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
```
Looking at your project, it seems that you're not using `pkg-config` to detect the ldflags and cflags. Here is the documentation about dune-configurator that will help fix this issue: https://dune.readthedocs.io/en/stable/dune-libs.html?highlight=dune-configurator#configurator

Also I've noticed that you included `conf-openimageio` in the source repository. It is recommended to not having it there and instead leave it to opam-repository to take care of it for you. Could you remove it from your source repository?

**Comment by kit-ty-kate** [comment]
> Looking at your project, it seems that you're not using `pkg-config` to detect the ldflags and cflags. Here is the documentation about dune-configurator that will help fix this issue: https://dune.readthedocs.io/en/stable/dune-libs.html?highlight=dune-configurator#configurator

Actually, that's also a bug in the Debian package itself, but in the future it would be wise to use `dune-configurator`

**Comment by zshipko** [comment]
Thank you, I appreciate all the help! I can definitely look into dune-configurator and remove that opam file.

**Comment by kit-ty-kate** [comment]
All green now. Thanks!

**Pull request merged**